common/workunit/workunit.cpp

```diff
@@ -5044,6 +5104,237 @@ bool CWorkUnitFactory::restoreWorkUnit(const char *base, const char *wuid, bool
+    {
+        StringAttr zapReportFileName, zapReportFilePath, zapReportPassword, importDir, user, wuid, scope;
+        SCMStringBuffer distributedAccessToken;
+        StringBuffer unzipDir, unzipDateTime;
+        Owned<IPTree> wuTree, graphProgressTree;
+
+        bool findZAPFile(const char *mask, bool optional, StringBuffer &fileName)
+        {
+            Owned<IFile> d = createIFile(unzipDir);
+            if (!d->exists())
+                throw MakeStringException(WUERR_InvalidUserInput, "%s not found.", unzipDir.str());
+
+            Owned<IDirectoryIterator> di = d->directoryFiles(mask, false, false);
+            if (!di->first())
+            {
+                if (optional)
+                    return false;
+                throw MakeStringException(WUERR_InvalidUserInput, "Failed to found %s in %s.", mask, unzipDir.str());
+            }
+
+            fileName.set(unzipDir).append(PATHSEPSTR);
+            di->getName(fileName);
+            return true;
+        }
+        void readWUXMLFileToPTree(Owned<IPTree> &tree, const char *pattern, bool optional)
+        {
+            StringBuffer fileName;
+            if (!findZAPFile(pattern, optional, fileName))
+                return;
+
+            tree.setown(createPTreeFromXMLFile(fileName));
+            if (!tree)
+                throw MakeStringException(WUERR_InvalidUserInput, "Failed to retrieving %s.", pattern);
+
+            Owned<IFile> f = createIFile(fileName);
+            f->remove();
+        }
+        void updateWUProcessLogAttrs(const char *xpath, const char *unzipDirWithIP, bool thorCluster)
+        { 
+            Owned<IPropertyTreeIterator> it = wuTree->getElements(xpath);
+            ForEach (*it)
+            {
+                StringBuffer process, logSpec;
+                IPropertyTree &proc = it->query();
+                proc.getName(process);
+                proc.getProp("@log", logSpec);
+                if (process.isEmpty() || logSpec.isEmpty())
+                    continue;
+
+                //Match the path and the name of the log files from ZAP report.
+                StringBuffer newLogAttr(unzipDirWithIP);
+                newLogAttr.append(process).append('_').append(pathTail(logSpec));
+                proc.setProp("@log", newLogAttr);
+                proc.setPropBool("@singleLog", true);
+
+                if (!thorCluster)
+                    continue;
+
+                StringBuffer slaveLogPattern;
+                slaveLogPattern.set(process).append("_");
+                if (proc.hasProp("@pattern"))
+                {
+                    proc.getProp("@pattern", slaveLogPattern);
+                }
+                else
+                {   //legacy WU
+                    getSlaveLogPatternForLegacyWU(logSpec, slaveLogPattern);
+                    proc.setPropInt("@max", findNumberOfThorSlavesFromSlaveLogs(unzipDirWithIP));
+                }
+                proc.setProp("@pattern", slaveLogPattern);
```

Comment:
Re. related to comment on line 5162.

I would expect you to be able to clear the existing entries, and call workunit.addProcess() to be called for each new log being imported.